device-selection-explainer.md

```diff
@@ -11,84 +11,86 @@ Feedback on this explainer is welcome via the issue tracker:
+However, the `deviceType` option, including the `"npu"` value, was later removed from `MLContextOptions` as part of a broader shift to make the context device-agnostic by default (see discussion around Issue #749 and PR #809).
+Alternative policies for error handling and fallback scenarios remained open questions.
 
-Later the need for explicit device selection support was challenged in [[MLContextOptions.deviceType seems unnecessary outside of conformance testing #749]](https://github.com/webmachinelearning/webnn/issues/749), with the main arguments also summarized in a W3C TPAC group meeting [presentation](https://lists.w3.org/Archives/Public/www-archive/2024Sep/att-0006/MLDeviceType.pdf). The main points were the following:
-- The [device type](https://webmachinelearning.github.io/webnn/#enumdef-mldevicetype) option is hard to standardize because of the heterogeneity of the compute units across various platforms, and even across their versions, for instance `"npu"` might not be a standalone option available, only a combined form of `"npu"` and `"cpu"`.
-- As for error management vs. fallback policies: fallback is preferable instead of failing, and implementations/the underlying platforms should determine the fallback type based on runtime information.
-- Implementation / browser / OS have better grasp of the system/compute/runtime/apps state than websites, and therefore control should be relished to them. For instance, if rendering performance degrades, the implementation/underlying platform can possibly fix it the best way, not the web app.
+Later, the need for explicit device selection support was challenged in [MLContextOptions.deviceType seems unnecessary outside of conformance testing #749](https://github.com/webmachinelearning/webnn/issues/749), with the main arguments also summarized in a W3C TPAC group meeting [presentation](https://lists.w3.org/Archives/Public/www-archive/2024Sep/att-0006/MLDeviceType.pdf). The main points were:
+- The [device type](https://webmachinelearning.github.io/webnn/#enumdef-mldevicetype) option is hard to standardize because of the heterogeneity of compute units across various platforms and even across their versions. For instance, `"npu"` might not be a standalone available option, only a combined form of `"npu"` and `"cpu"`.
+- As for error management vs. fallback policies: fallback is preferable to failing, and implementations/the underlying platforms should determine the fallback type based on runtime information.
+- Implementations, browsers, or the OS have a better grasp of the system, compute, runtime, and application state than websites, and therefore control should be relinquished to them. For instance, if rendering performance degrades, the implementation/underlying platform can possibly fix it the best way, not the web app.
+This led to the changes implemented in PR #809.
 
 ## Key use cases and requirements
 
 ### Device Preference Use Cases
```

Comment:
Feedback https://github.com/webmachinelearning/webnn/issues/815#issuecomment-2962608053 from Google Meet suggests there is an "accelerated execution" use case:

>Thanks! For our highest quality model, we have a notion that accelerated (GPU and (we think?) NPU) execution is required for most of our users' hardware, and that preference is not listed yet.

I suggest we add "accelerated execution" to this list.

The feedback further suggests it is an implementation detail what processing unit(s) are used. The expectation is the system is capable of "realtime execution" with this preference.

Reply:
I'd add "accelerated execution" as its own item to the list, not refer to any specific device types in its description for future-proofing. Any processing unit can have instructions that accelerate neural network computations.

I'd consider Device Preference Use Cases a documentation of wishes from API consumers, and these wishes are to be validated against implementability realities when translated to normative spec prose.

Reply:
This "accelerated execution" requirement is documented in use case: "1. Pre-download capability check".